src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
+pub trait FileRepository {
+    fn read_template_file(&self, path: &TemplateLocation) -> Result<String>;
```

Comment:
My goal with caching is to reduce disk access + parse time when batch processing templates like in cobalt.

Could you elaborate on your idea for `LiquidRendering -> TemplateFactory -> TemplateRepository`?  What is the role of each, how do they compose, etc?